### ISSUE TITLE
Animation's `remover` parameter does not work

### ISSUE THREAD

**Issue by WampyCakes** [ManimCommunity/manim#1582]
## Description of bug / unexpected behavior

Setting `remover=true` on the Animation class should remove the mobject. It doesn't.

## How to reproduce the issue

<details><summary>Code for reproducing the problem</summary>

```py
Paste your code here.
```

</details>

## Additional media files

<details><summary>Images/GIFs</summary>

</details>

## Logs
<details><summary>Terminal output</summary>

```
PASTE HERE OR PROVIDE LINK TO https://pastebin.com/ OR SIMILAR
```

</details>

## System specifications

<details><summary>System Details</summary>

- OS (with version, e.g Windows 10 v2004 or macOS 10.15 (Catalina)):
- RAM:
- Python version (`python/py/python3 --version`):
- Installed modules (provide output from `pip list`):
```
PASTE HERE
```
</details>

<details><summary>LaTeX details</summary>

+ LaTeX distribution (e.g. TeX Live 2020):
+ Installed LaTeX packages:

</details>

<details><summary>FFMPEG</summary>

Output of `ffmpeg -version`:

```
PASTE HERE
```
</details>

## Additional comments

See https://github.com/ManimCommunity/manim/pull/1374#discussion_r620557421

**Comment by WampyCakes** [comment]
Upon fixing this bug, if #1374 is merged, the implementation of the `Remove` animation should also be updated -- easy change.

**Comment by WampyCakes** [comment]
Upon further investigation, the paramter works when proceeded by a `self.wait()` or another animation. I think this may be a deeper timing related issue with actually finishing an animation.

**Comment by huguesdevimeux** [comment]
This is weird, I seeing the code I don't see any problem with the `remover` attr for Animation. 
Its logic is [there](https://github.com/ManimCommunity/manim/blob/6e4b02a4d1250e462a88489c826dc5af795947ca/manim/animation/animation.py#L197-L209) 

However, for `Wait`, it does not [work](https://github.com/ManimCommunity/manim/blob/6e4b02a4d1250e462a88489c826dc5af795947ca/manim/animation/animation.py#L478-L479)

**Comment by WampyCakes** [comment]
Yes, the functionality is there. It just doesn't work unless the animation with a remover is followed by either a wait or another animation. So it seems to be not an issue of animation, but `clean_up_from_scene` being called incorrectly internally. IIRC there are issues with completing the last frames of an animation/timing, so I think it _could maybe_ be related.

**Comment by PhotonSpheres** [comment]
Can you give an example where it doesn't work? Because in this example here:
```py
class RemoverTest(Scene):
    def construct(self):
        sq = Square()
        self.play(
            Write(sq),
            remover=True
        )
        print(self.mobjects)
```
it seems to work without calling `self.wait()` or another animation. I still see the square at the end of the animation, I agree with you that this is due to the animation not fully rendering. But the scene remains empty after the animation.

**Comment by WampyCakes** [comment]
I didn't even know you could pass `remover` to `play()`.

Anyhow, this _should_ work but doesn't:
```py
def construct(self):
    s = Square()
    self.add(s)
    self.play(Animation(s, remover=True))
```

**Comment by PhotonSpheres** [comment]
I see, if I do this:
```py
class RemoverTest3(Scene):
    def construct(self):
        s = Square()
        print(self.mobjects)
        self.add(s)
        print(self.mobjects)
        self.play(Animation(s, remover=True))
        print(self.mobjects)
```
The output of my console is still
```
[]
[Square]
[]
```
Doesn't this mean that the square is removed from the scene internally? Or are we only talking about that at the end of the animation the rendered output file has still the square visible in the last frame? Sorry for the confusion, was just curious and eager to help.

**Comment by WampyCakes** [comment]
Interesting. I guess that kind of explains why this removes it:
```py
def construct(self):
    s = Square()
    self.add(s)
    self.play(Animation(s, remover=True))
    self.wait()
```
but it won't without the Wait.

I expect `remover` to actually remove the mobject from the scene (as more than just myself thought in #1374). I think this is also expected behavior since with the Wait it _is_ removed from the scene and not just internally.

**Comment by PhotonSpheres** [comment]
I agree `remover` should do that. But I guess this kinda proves that there is probably nothing wrong with `remover` or `clean_up_from_scene` but rather the problem is inherited by the fact that scenes do not fully render? As in issue #183 discussed?

**Comment by WampyCakes** [comment]
Good chance. It's a shame that issue has lasted this long :/
